source/module_io/write_elf.cpp

```diff
@@ -0,0 +1,168 @@
+#include "write_elf.h"
+#include "module_io/cube_io.h"
+
+namespace ModuleIO
+{
+void write_elf(
+#ifdef __MPI
+    const int& bz,
+    const int& nbz,
+#endif
+    const std::string& out_dir,
+    const int& istep,
+    const int& nspin,
+    const double* const* rho,
+    const double* const* tau,
+    ModulePW::PW_Basis* rho_basis,
+    const UnitCell* ucell_)
+{
+    std::vector<std::vector<double>> elf(nspin, std::vector<double>(rho_basis->nrxx, 0.));
+    // 1) calculate the kinetic energy density of vW KEDF
+    std::vector<std::vector<double>> tau_vw(nspin, std::vector<double>(rho_basis->nrxx, 0.));
+    for (int is = 0; is < nspin; ++is)
+    {
+        std::vector<std::vector<double>> nabla_rho(3, std::vector<double>(rho_basis->nrxx, 0.));
```

Comment:
is this the laplacian of rho? I think the name like "laplacian_rho" will be better than nabla because the former is a physical name

Reply:
It is the gradient of rho $\nabla\rho$, I have renamed it as "gradient_rho"